pkgs/applications/radio/jtdx/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, fetchFromGitHub, asciidoc, asciidoctor, autoconf, automake, cmake,
+  docbook_xsl, fftw, fftwFloat, gfortran, libtool, libusb1, qtbase,
+  qtmultimedia, qtserialport, qttools, boost, texinfo, wrapQtAppsHook, jtdxhamlib, libsForQt5 }:
+
+stdenv.mkDerivation rec {
+  pname = "jtdx";
+  version = "2.2.159";
+
+  src = fetchFromGitHub {
+    owner = "jtdx-project";
+    repo = "jtdx";
+    rev = "159";
+    sha256 = "sha256-5KlFBlzG3hKFFGO37c+VN+FvZKSnTQXvSorB+Grns8w=";
+  };
+
+  nativeBuildInputs = [
+    asciidoc asciidoctor autoconf automake cmake docbook_xsl gfortran libtool
+    qttools texinfo wrapQtAppsHook libsForQt5.qtwebsockets
+  ];
+  buildInputs = [ fftw fftwFloat libusb1 qtbase qtmultimedia qtserialport boost jtdxhamlib ];
```

Comment:
Is it absolutely necessary to use the jtdxhamlib? AIUI, this was forked due to changes in hamlib 4 - have you tried this with hamlib 3? Both hamlib3 and hamlib 4 are in nixpkgs.